src/components/Checkbox.js

```diff
@@ -1,65 +1,23 @@
-import React, { useCallback } from 'react'
-import propTypes from 'prop-types'
-import { Field, Checkbox as CoreCheckbox, Help } from '@dhis2/ui-core'
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import { CheckboxField } from '@dhis2/ui-core'
 
-import { FieldAdapter, adapterComponentProps } from './FieldAdapter.js'
+import { createToggleChangeHandler, normalizeProps } from './shared/helpers.js'
+import { fieldRenderProps } from './shared/propTypes.js'
 
-const Checkbox = ({
-    label,
-    required,
-    disabled,
-    name,
-    value,
-    checkedValue,
-    onChange,
-    error,
-    warning,
-    valid,
-    helpText,
-    errorText,
-}) => (
-    <Field>
-        <CoreCheckbox
-            required={required}
-            name={name}
-            disabled={disabled}
-            value={checkedValue}
-            label={label}
-            checked={!!value}
-            onChange={onChange}
-            error={error}
-            warning={warning}
-            valid={valid}
-        />
-        {helpText && <Help>{helpText}</Help>}
-        {error && errorText && <Help error>{errorText}</Help>}
-    </Field>
-)
+const Checkbox = props => {
+    const { value, checkedValue, ...rest } = normalizeProps(
+        props,
+        createToggleChangeHandler(props.input.onChange)
+    )
 
-Checkbox.propTypes = {
-    ...adapterComponentProps,
-    checkedValue: propTypes.string,
+    return <CheckboxField {...rest} checked={!!value} value={checkedValue} />
 }
 
-// If the input has a value (checkedValue prop) the form value is: checkedValue || ''
-// Otherwise the form-value is: true || false
-const getValue = ({ checked, value }) => {
-    if (value) {
-        return checked ? value : ''
-    } else {
-        return !!checked
-    }
+Checkbox.propTypes = {
+    ...fieldRenderProps,
+    ...CheckboxField.propTypes,
```

Comment:
I'd list the props directly here. It requires more work initially, but they'll be lintable and explicit.

Reply:
We discussed to classify this as out-of-scope for now. I have created an issue so we don't forget to address this in the future: https://github.com/dhis2/ui-forms/issues/153